### PR TITLE
Group model

### PR DESCRIPTION
**Pull request by Floury-Fabio**
Related ticket : #218

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -12,7 +12,7 @@ gem 'jbuilder', '~> 2.5'
 gem 'jquery-rails'
 gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 3.12'
-gem 'rails', '~> 5.2', '>= 5.2.4'
+gem 'rails', '~> 5.2'
```
Comment (Floury-Fabio) [inline Gemfile:15]: maj

File: app/admin/groups.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# rubocop: disable Metrics/BlockLength
```
Comment (Floury-Fabio) [inline app/admin/groups.rb]: rm disable rubocop

File: app/admin/groups.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      end
+      table_for group.members do
+        column 'members' do |member|
+          link_to "#{member.last_name} #{member.first_name}", [:admin, member]
+        end
+      end
+    end
+  end
+end
+# rubocop: enable Metrics/BlockLength
```
Comment (Floury-Fabio) [inline app/admin/groups.rb]: rm enable rubocop

File: app/admin/members.rb [hunk 4]
```diff
@@ -2,17 +2,18 @@
 
 # rubocop: disable Metrics/BlockLength
 ActiveAdmin.register Member do
-  permit_params :email, :password, :encrypted_password, :first_name, :last_name,
-                :biography, :phone_number, :role, :moderator, :group, :confirmed_at,
-                :password_confirmation, :cash_register_proficiency, :register_id
+  includes :groups
+  permit_params :email, :password, :encrypted_password, :first_name, :last_name, :biography, :phone_number, :role,
```
Comment (Floury-Fabio) [inline app/admin/members.rb]: one params per line

File: app/admin/members.rb [hunk 5]
```diff
@@ -2,17 +2,18 @@
...
+                group_ids: []
 
   index do
     selectable_column
     column :first_name
     column :last_name
     column :role
     column('3 heures faites?') { |member| member.worked_three_hours?(Date.current) }
-    column(:group) { |member| Member.human_enum_name(:group, member.group) }
+    column(:group) { |member| member.groups.map(&:name).join(', ') }
```
Comment (Floury-Fabio) [inline app/admin/members.rb]: add link to groups (view `auto_link` helper)

File: app/admin/members.rb [hunk 6]
```diff
@@ -29,14 +30,28 @@
...
   end
 
+  show do
+    attributes_table_for resource do
+      default_attribute_table_rows.each do |field|
+        row field
+      end
+      table_for member.groups do
+        column 'groups' do |group|
+          link_to Arbre::Context.new { (status_tag class: 'important', label: group.name) }, [:admin, group]
```
Comment (Floury-Fabio) [inline app/admin/members.rb:58]: test with auto_link

File: app/admin/members.rb [hunk 7]
```diff
@@ -29,14 +30,28 @@
...
+        end
+      end
+    end
+  end
+
   form do |f|
     f.inputs :first_name, :last_name, :email, :phone_number, :role, :moderator,
-             :group, :cash_register_proficiency, :register_id, :biography
+             :cash_register_proficiency, :register_id, :biography
```
Comment (Floury-Fabio) [inline app/admin/members.rb]: one params per line

File: app/models/application_record.rb [hunk 8]
```diff
@@ -8,4 +8,10 @@ def self.human_enum_name(enum_name, enum_value)
 
     I18n.t("activerecord.attributes.#{model_name.i18n_key}.#{enum_name.to_s.pluralize}.#{enum_value}")
   end
+
+  def self.human_model_name(value)
+    return if value.nil?
```
Comment (Floury-Fabio) [inline app/models/application_record.rb]: rm this

File: app/models/group.rb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+#
+# Table name: missions
+#
+#  id                       :bigint(8)        not null, primary key
+#  name                     :string           not null
+#  group_manager_mail       :string
+
+# The members contribute to one or many groups. Each groups have a specific function in the association.
+class Group < ApplicationRecord
+  belongs_to :manager, class_name: :Member, inverse_of: 'managed_group', optional: true
```
Comment (Floury-Fabio) [inline app/models/group.rb]: manager has_many managed groups

File: spec/models/member_spec.rb [hunk 10]
```diff
@@ -71,7 +67,14 @@
           .class_name('Mission').with_foreign_key('author_id')
           .dependent(:nullify)
       }
+      it {
+        expect(instance).to have_one(:managed_group)
+          .class_name('Group').with_foreign_key('manager_id')
```
Comment (Floury-Fabio) [inline spec/models/member_spec.rb]: has_many

File: spec/models/member_spec.rb [hunk 11]
```diff
@@ -71,7 +67,14 @@
...
+          .dependent(:nullify)
+      }
       it { is_expected.to have_many(:missions).through(:enrollments) }
+      it { is_expected.to have_many(:groups).through(:group_members) }
+      it { is_expected.to accept_nested_attributes_for(:groups).allow_destroy(true) }
```
Comment (Floury-Fabio) [inline spec/models/member_spec.rb]: change expected.to accept_nested_attributes_for(:groups) to 
expected.to accept_nested_attributes_for(:group_members)

File: spec/models/group_spec.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+#  group_manager_mail       :string
+
+require 'rails_helper'
+
+RSpec.describe Group, type: :model do
+  describe 'Model instanciation' do
+    subject { described_class.new }
+
+    describe 'Database' do
+      it { is_expected.to have_db_column(:name).of_type(:string).with_options(null: false) }
```
Comment (Floury-Fabio) [inline spec/models/group_spec.rb]: rm database test

File: spec/models/group_spec.rb [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    end
+
+    describe 'validations' do
+      it { is_expected.to validate_presence_of(:name) }
+      it { validate_uniqueness_of(:name).case_insensitive }
+    end
+
+    describe 'associations' do
+      it { is_expected.to have_many(:members).through(:group_members) }
+      it { is_expected.to belong_to(:manager).class_name('Member').inverse_of('managed_group').optional }
```
Comment (Floury-Fabio) [inline spec/models/group_spec.rb]: change belong_to to have_many

File: spec/models/group_member_spec.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
```
Comment (Floury-Fabio) [inline spec/models/group_member_spec.rb:2]: add belongs to member
add belongs to group

File: app/models/member.rb [hunk 15]
```diff
@@ -47,6 +46,11 @@ class Member < ApplicationRecord
   has_one :address, dependent: :destroy
   accepts_nested_attributes_for :address, reject_if: :all_blank, allow_destroy: true
 
+  has_one :managed_group, class_name: 'Group', inverse_of: :manager, foreign_key: 'manager_id', dependent: :nullify
```
Comment (Floury-Fabio) [inline app/models/member.rb]: change has_one to has_many

File: app/models/member.rb [hunk 16]
```diff
@@ -47,6 +46,11 @@ class Member < ApplicationRecord
...
+  has_many :group_members, dependent: :destroy
+  has_many :groups, through: :group_members
+  accepts_nested_attributes_for :groups, allow_destroy: true
```
Comment (Floury-Fabio) [inline app/models/member.rb]: see test comments

File: app/presenters/group_presenter.rb [hunk 17]
```diff
@@ -0,0 +1,19 @@
+class GroupPresenter
+  def initialize(group)
+    @group = group
+  end
+
+  def convert_name
+    @group.name.split.join('_')
```
Comment (Floury-Fabio) [inline app/presenters/group_presenter.rb:12]: replace with `#underscore`

File: app/views/members/index.html.erb [hunk 18]
```diff
@@ -16,10 +16,11 @@
           </a>
         </li>
 
-        <% Member.groups.each do |group_name, _| %>
+        <% @groups.each do |group| %>
+          <% presenter = GroupPresenter.new(group) %>
           <li class="nav-item">
-            <a class="nav-link" href="#<%= group_name %>-group" data-toggle="tab" role="tab">
-              <%= Member.human_enum_name(:group, group_name).humanize %>
+            <a class="nav-link" href="#<%= presenter.convert_name %>-group" data-toggle="tab" role="tab">
```
Comment (Floury-Fabio) [inline app/views/members/index.html.erb]: `String.underscore`

File: app/presenters/group_presenter.rb [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def manager
+    return '' if @group.manager.nil?
+    return "#{I18n.t('activerecord.attributes.group.manager')}: #{@group.manager.first_name} #{@group.manager.last_name}" unless @group.manager.nil?
+  end
+
+  def manager_id
+    return 0 if @group.manager.nil?
+    return @group.manager.id unless @group.manager.nil?
```
Comment (Floury-Fabio) [inline app/presenters/group_presenter.rb]: rm this and
manage link presentation with tag.a

File: app/presenters/group_presenter.rb [hunk 20]
```diff
@@ -0,0 +1,19 @@
+class GroupPresenter
+  def initialize(group)
+    @group = group
+  end
+
+  def convert_name
+    @group.name.split.join('_')
+  end
+
+  def manager
```
Comment (Floury-Fabio) [inline app/presenters/group_presenter.rb]: replace `#manager` by `#manager_full_name`

File: app/views/members/index.html.erb [hunk 21]
```diff
@@ -32,11 +33,14 @@
           </div> 
         </section>
 
-        <% Member.groups.each do |group_name, _| %>
-          <section class="tab-pane fade" id="<%= group_name %>-group" role="tabpanel" aria-labelledby="<%= group_name %>-group-tab">
+        <% @groups.each do |group| %>
+          <% presenter = GroupPresenter.new(group) %>
+          <section class="tab-pane fade" id="<%= presenter.convert_name %>-group" role="tabpanel" aria-labelledby="<%= presenter.convert_name %>-group-tab">
```
Comment (Floury-Fabio) [inline app/views/members/index.html.erb]: `String.underscore`

File: app/views/members/index.html.erb [hunk 22]
```diff
@@ -32,11 +33,14 @@
...
+            <div> <%= link_to presenter.manager, member_path(presenter.manager_id) %> </div>
```
Comment (Floury-Fabio) [inline app/views/members/index.html.erb]: Replace with presenter link helper

File: app/views/members/show.html.erb [hunk 23]
```diff
@@ -27,9 +27,10 @@
               </h6>
               <br/>
 
-              <% if @member.group %>
+              <% if @member.groups.any? %>
                 <h6 class="heading heading-light heading-sm strong-300">
-                  Groupe <%= Member.human_enum_name(:group, @member.group) %>
+                  Groupe:
```
Comment (Floury-Fabio) [inline app/views/members/show.html.erb]: replace `groupe` by `groupes`

File: config/locales/models/group.en.yml [hunk 24]
```diff
@@ -0,0 +1,11 @@
+en:
+  activerecord:
+    models:
+      group:
+        one: Group
+        other: Groups
+    attributes:
+      group:
+        name: name
+        number_of_members: number of members
```
Comment (Floury-Fabio) [inline config/locales/models/group.en.yml]: replace `number_of_members` by `members_count`
rm redondante I18n key

File: config/locales/models/group.fr.yml [hunk 25]
```diff
@@ -0,0 +1,11 @@
+fr:
+  activerecord:
+    models:
+      group:
+        one: Groupe
+        other: Groupes
+    attributes:
+      group:
+        name: nom
+        number_of_members: nombre de membres
```
Comment (Floury-Fabio) [inline config/locales/models/group.fr.yml]: replace number_of_members by members_count

File: create_tempo_table.txt [hunk 26]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE tempo_group_table ( number int, name varchar(255));
```
Comment (Floury-Fabio) [inline create_tempo_table.txt]: try to replace `.txt` to `.sql`

File: db/seeds.rb [hunk 27]
```diff
@@ -5,55 +5,85 @@
...
+Group.destroy_all
 Member.destroy_all
 Productor.destroy_all
 Document.destroy_all
 Thredded::Messageboard.destroy_all
 Thredded::MessageboardGroup.destroy_all
 
 9.times do |index|
-  FactoryBot.create :member, :admin, :group,
+  FactoryBot.create :member, :admin,
```
Comment (Floury-Fabio) [inline db/seeds.rb]: one params per line

File: db/seeds.rb [hunk 28]
```diff
@@ -5,55 +5,85 @@
...
                     address: FactoryBot.create(:address),
                     email: "admin#{index}@admin.com"
 end
-FactoryBot.create :member, :super_admin, :group,
+
+FactoryBot.create :member, :super_admin,
```
Comment (Floury-Fabio) [inline db/seeds.rb]: one params per line

File: db/seeds.rb [hunk 29]
```diff
@@ -5,55 +5,85 @@
...
+
+Rails.logger.info 'groups seeded'
+
+10.times do
   FactoryBot.create :productor, address: FactoryBot.create(:address, :coordinates)
   FactoryBot.create :productor, local: true,
-                                address: FactoryBot.create(:address, :coordinates)
-}
-puts "Productors seeded"
+    address: FactoryBot.create(:address, :coordinates)
```
Comment (Floury-Fabio) [inline db/seeds.rb]: align argument

File: db/seeds.rb [hunk 30]
```diff
@@ -5,55 +5,85 @@
...
+FactoryBot.create :group, name: 'orders_management'
+FactoryBot.create :group, name: 'it'
+
+Member.all.each do |member|
+  rand(4).times do
+    FactoryBot.create :group_member, member: member, group: Group.all.sample
+  end
+end
+
+Rails.logger.info 'groups seeded'
```
Comment (Floury-Fabio) [inline db/seeds.rb]: Initialize rails logger to output in terminal

File: db/seeds.rb [hunk 31]
```diff
@@ -5,55 +5,85 @@
...
-10.times {
+Rails.logger.info 'Productors seeded'
+
+10.times do
   FactoryBot.create :mission, members: Member.all.sample(rand(0..8)),
-                              addresses: FactoryBot.create_list(:address, rand(1..2)),
-                              author: Member.all.sample
-}
-puts "Missions seeded"
+  addresses: FactoryBot.create_list(:address, rand(1..2)),
```
Comment (Floury-Fabio) [inline db/seeds.rb]: align arguments

File: db/seeds.rb [hunk 32]
```diff
@@ -5,55 +5,85 @@
...
   document.save!
 end
-puts "Documents seeded"
+
+Rails.logger.info 'Documents seeded'
 
 FactoryBot.create_list :messageboard_group, 2
 FactoryBot.create_list :messageboard, 6,
-                       messageboard_group_id: Thredded::MessageboardGroup.all.sample.id
+  messageboard_group_id: Thredded::MessageboardGroup.all.sample.id
```
Comment (Floury-Fabio) [inline db/seeds.rb]: align argument

File: db/seeds.rb [hunk 33]
```diff
@@ -5,55 +5,85 @@
...
 
-18.times {
+18.times do
   FactoryBot.create :topic, with_posts: rand(1..4),
-                            messageboard: Thredded::Messageboard.all.sample
-}
-puts "Forum seeded"
+  messageboard: Thredded::Messageboard.all.sample
```
Comment (Floury-Fabio) [inline db/seeds.rb]: align argument

File: group_migrate.txt [hunk 34]
```diff
@@ -0,0 +1,13 @@
+
+START TRANSACTION;
+
+INSERT INTO group_members
+  (group_id, member_id, created_at, updated_at)
+SELECT groups.id AS group_id, members.id AS member_id, NOW(), NOW()
+FROM members
+INNER JOIN tempo_group_table
+ON members.group = tempo_group_table.number
```
Comment (Floury-Fabio) [inline group_migrate.txt]: indent with logical steps

File: insert_group_in_group_table.txt [hunk 35]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO groups (name, created_at, updated_at)
```
Comment (Floury-Fabio) [inline insert_group_in_group_table.txt]: rename file name into "insert existing groups in new groups table"

File: lib/tasks/migrate_for_new_group_management.rake [hunk 36]
```diff
@@ -0,0 +1,35 @@
+desc 'migrate for new group management'
+task migration_sql: :environment do
+  if ActiveRecord::Base.connection.table_exists?(:tempo_group_table)
+    ActiveRecord::Base.connection.execute('DROP TABLE tempo_group_table')
+  end
+
+  Group.destroy_all
+  file = File.open('insert_group_in_group_table.txt')
```
Comment (Floury-Fabio) [inline lib/tasks/migrate_for_new_group_management.rake]: rename file name into "insert existing groups in new groups table"

File: lib/tasks/migrate_for_new_group_management.rake [hunk 37]
```diff
@@ -0,0 +1,35 @@
...
+
+  file = File.open('create_tempo_table.txt')
+  ActiveRecord::Base.connection.execute(file.read)
+  file.close
+
+  file = File.open('group_migrate.txt')
+  ActiveRecord::Base.connection.execute(file.read)
+  file.close
+
+  verification
```
Comment (Floury-Fabio) [inline lib/tasks/migrate_for_new_group_management.rake]: rename `#verification`  to `#check_data_consistency`

File: spec/features/factories_test.rb [hunk 38]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'spec_helper'
```
Comment (Floury-Fabio) [inline spec/features/factories_test.rb:4]: rm this line, `spec_helper` is already required by `rails_helper`

File: app/admin/members.rb (hunk 5 above)
Comment (Floury-Fabio) [reply app/admin/members.rb]: Done

File: app/admin/groups.rb (hunk 2 above)
Comment (Floury-Fabio) [reply app/admin/groups.rb]: done

File: app/admin/groups.rb (hunk 3 above)
Comment (Floury-Fabio) [reply app/admin/groups.rb]: done

File: app/admin/members.rb (hunk 4 above)
Comment (Floury-Fabio) [reply app/admin/members.rb]: done

File: app/admin/members.rb (hunk 6 above)
Comment (Floury-Fabio) [reply app/admin/members.rb:58]: it didn´t work

File: app/admin/members.rb (hunk 7 above)
Comment (Floury-Fabio) [reply app/admin/members.rb]: done

File: app/models/application_record.rb (hunk 8 above)
Comment (Floury-Fabio) [reply app/models/application_record.rb]: done

File: spec/models/member_spec.rb (hunk 11 above)
Comment (Floury-Fabio) [reply spec/models/member_spec.rb]: We want see and edit relation with groups and not with group_members. We must have to one checkbox per group and for that we need `accept_nested_attributes_for(:groups)`. Active admin create and delete group_members automicaly according to the selection of user.

File: spec/models/group_spec.rb (hunk 12 above)
Comment (Floury-Fabio) [reply spec/models/group_spec.rb]: done

File: app/models/group.rb (hunk 9 above)
Comment (Floury-Fabio) [reply app/models/group.rb]: done

File: spec/models/member_spec.rb (hunk 10 above)
Comment (Floury-Fabio) [reply spec/models/member_spec.rb]: done

File: spec/models/member_spec.rb (hunk 11 above)
Comment (Floury-Fabio) [reply spec/models/member_spec.rb]: Finally nested resource is useless in our  configuration.

File: spec/models/group_spec.rb (hunk 13 above)
Comment (Floury-Fabio) [reply spec/models/group_spec.rb]: done

File: spec/models/group_member_spec.rb (hunk 14 above)
Comment (Floury-Fabio) [reply spec/models/group_member_spec.rb:2]: done

File: app/models/member.rb (hunk 15 above)
Comment (Floury-Fabio) [reply app/models/member.rb]: done

File: app/models/member.rb (hunk 16 above)
Comment (Floury-Fabio) [reply app/models/member.rb]: done

File: app/presenters/group_presenter.rb (hunk 17 above)
Comment (Floury-Fabio) [reply app/presenters/group_presenter.rb:12]: done

File: app/views/members/index.html.erb (hunk 18 above)
Comment (Floury-Fabio) [reply app/views/members/index.html.erb]: done

File: app/presenters/group_presenter.rb (hunk 19 above)
Comment (Floury-Fabio) [reply app/presenters/group_presenter.rb]: done with import `link_to` and `_path`

File: app/presenters/group_presenter.rb (hunk 20 above)
Comment (Floury-Fabio) [reply app/presenters/group_presenter.rb]: done

File: app/views/members/index.html.erb (hunk 21 above)
Comment (Floury-Fabio) [reply app/views/members/index.html.erb]: done

File: app/views/members/index.html.erb (hunk 22 above)
Comment (Floury-Fabio) [reply app/views/members/index.html.erb]: done

File: config/locales/models/group.en.yml (hunk 24 above)
Comment (Floury-Fabio) [reply config/locales/models/group.en.yml]: done

File: config/locales/models/group.fr.yml (hunk 25 above)
Comment (Floury-Fabio) [reply config/locales/models/group.fr.yml]: done

File: create_tempo_table.txt (hunk 26 above)
Comment (Floury-Fabio) [reply create_tempo_table.txt]: done

File: db/seeds.rb (hunk 27 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 28 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 29 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 30 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 31 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 32 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: db/seeds.rb (hunk 33 above)
Comment (Floury-Fabio) [reply db/seeds.rb]: done

File: group_migrate.txt (hunk 34 above)
Comment (Floury-Fabio) [reply group_migrate.txt]: done

File: insert_group_in_group_table.txt (hunk 35 above)
Comment (Floury-Fabio) [reply insert_group_in_group_table.txt]: done

File: lib/tasks/migrate_for_new_group_management.rake (hunk 36 above)
Comment (Floury-Fabio) [reply lib/tasks/migrate_for_new_group_management.rake]: done

File: app/admin/groups.rb [hunk 39]
```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+ActiveAdmin.register Group do
+  permit_params :name, manager_ids: []
+
+  index do
+    selectable_column
+    column :name
+    column(:managers) do |managed_group|
```
Comment (Mate2xo) [inline app/admin/groups.rb]: Rename `managed_group` to `group`

File: app/admin/groups.rb [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+      group_links = managed_group.managers.map { |manager| auto_link manager }
```
Comment (Mate2xo) [inline app/admin/groups.rb]: rename `group_links` to `managers_list`

File: app/admin/groups.rb [hunk 41]
```diff
@@ -0,0 +1,40 @@
...
+      safe_join group_links, ', '
+    end
+    column(I18n.t('activerecord.attributes.group.number_of_members')) { |group| group.members.size }
```
Comment (Mate2xo) [inline app/admin/groups.rb]: use I18n lookup shortcut `t('.members_count')`

File: app/admin/groups.rb [hunk 42]
```diff
@@ -0,0 +1,40 @@
...
+    actions
+  end
+
+  show do
+    attributes_table_for resource do
+      default_attribute_table_rows.each do |field|
+        row field
+      end
+      table_for group.members do
+        column 'members' do |member|
```
Comment (Mate2xo) [inline app/admin/groups.rb]: use I18n :
`Member.model_name.human`

File: app/admin/groups.rb [hunk 43]
```diff
@@ -0,0 +1,40 @@
...
+          link_to "#{member.last_name} #{member.first_name}", [:admin, member]
+        end
+      end
+      table_for group.managers do
+        column 'managers' do |manager|
```
Comment (Mate2xo) [inline app/admin/groups.rb]: I18n:
`resource.class.human_attribute_name(:managers)`

File: app/admin/members.rb [hunk 44]
```diff
@@ -29,14 +44,35 @@
     column("3 heures de #{l 1.month.ago, format: '%B'}?") { |member| member.worked_three_hours?(1.month.ago) }
     column("3 heures de #{l Date.current, format: '%B'}?") { |member| member.worked_three_hours?(Date.current) }
     column("3 heures de #{l 1.month.from_now, format: '%B'}?") { |member| member.worked_three_hours?(1.month.from_now) }
-    column(:group) { |member| Member.human_enum_name(:group, member.group) }
+    column(:group) { |member| member.groups.map(&:name).join(', ') }
```
Comment (Mate2xo) [inline app/admin/members.rb:46]: Remove duplicate column

File: app/presenters/group_presenter.rb (hunk 17 above)
Comment (Mate2xo) [reply app/presenters/group_presenter.rb:12]: https://devdocs.io/rails~5.2/string#method-i-underscore (doesn't replaces spaces)

File: spec/factories/group_managers.rb [hunk 45]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :group_manager do
+    managed_group { nil }
+    manager { nil }
```
Comment (Mate2xo) [inline spec/factories/group_managers.rb]: Check if `belongs_to` validations are effective (?)

File: spec/factories/group_members.rb [hunk 46]
```diff
@@ -0,0 +1,4 @@
+FactoryBot.define do
+  factory :group_member do
+  end
```
Comment (Mate2xo) [inline spec/factories/group_members.rb:3]: Same as above, check if foreign keys validations are ok

**Review by Mate2xo — CHANGES_REQUESTED** [review]

**Comment by Floury-Fabio** [comment]
https://github.com/Mate2xo/Creeons_la_coop/issues/218

**Review by Mate2xo — APPROVED** [review]

**Pull request merged**